courier/test/persistence.go

```diff
@@ -126,12 +126,60 @@ func TestPersister(ctx context.Context, newNetworkUnlessExisting NetworkWrapper,
 			assert.Equal(t, messages[len(messages)-1].ID, ms[0].ID)
 
 			t.Run("on another network", func(t *testing.T) {
-				_, p := newNetwork(t, ctx)
-				ms, tc, _, err := p.ListMessages(ctx, filter, []keysetpagination.Option{})
+				nid1, p1 := newNetwork(t, ctx)
+				ms, tc, _, err := p1.ListMessages(ctx, filter, []keysetpagination.Option{})
 
 				require.NoError(t, err)
 				require.Len(t, ms, 0)
 				require.Equal(t, int64(0), tc)
+
```

Comment:
This is what a failure of the test case looks like: https://github.com/ory/kratos/actions/runs/3856498251/jobs/6572792385#step:15:3563

Reply:
That's terrifying.

Reply:
Yup, luckily this was _very_ unlikely, as there had to have been three preconditions: 

1. two messages with a timestamp collision (microsecond precision)
2. the first message needed to be a page token
3. the second message needed to have a UUID that had a higher "string" value (e.g. 1: "00...", 2: "11...")